File: src/Submission/RegOS.Submission.Application/Queries/GetApplicationContacts/ApplicationContacts.cs
namespace RegOS.Submission.Application.Queries.GetApplicationContacts;

/// <summary>
/// Who currently speaks for an application — <b>derived, never stored</b>
/// (ADR-048).
/// </summary>
/// <param name="AsOfSequenceNumber">
/// The sequence these were read from. Null when the application has published
/// nothing: before the first filing there is nobody named on a filing, and that
/// is an absence of a filing rather than missing data.
/// </param>
public sealed record ApplicationContacts(
    int? AsOfSequenceNumber,
    IReadOnlyList<ApplicationContact> Contacts);

public sealed record ApplicationContact(
    Guid ContactId,
    string ContactName,
    string? ContactTitle,
    string OrganizationName,
    Guid RoleId,
    string RoleName);

File: web/regos-web/src/features/regulatory/submissions/types/ApplicationContacts.ts
/**
 * Who currently speaks for an application — **derived, never stored**.
 *
 * There is deliberately no application-level contact model (ADR-048): under the
 * cumulative model the latest published sequence *is* the current regulatory
 * state, so a stored copy could only differ from it by being stale.
 */
export interface ApplicationContacts {
  /** Null when nothing has been published — an absence of a filing. */
  asOfSequenceNumber: number | null;
  contacts: ApplicationContact[];
}

export interface ApplicationContact {
  contactId: string;
  contactName: string;
  contactTitle: string | null;
  organizationName: string;
  roleId: string;
  roleName: string;
}
